setup.sh

```diff
@@ -179,20 +179,24 @@ if ! osx; then
     PYTHON+="${PYVER}"
 fi
 
-# Create Python virtualenv
+# Install Poetry
+if ! command -v poetry &> /dev/null; then
+    echo "Poetry not found. Installing Poetry..."
+    curl -sSL https://install.python-poetry.org | python3 -
+    export PATH="$HOME/.local/bin:$PATH"
```

Comment:
I'm wondering if instead of exporting `PATH` here we should instead do `POETRY="$HOME/.local/bin/poetry`, and then use `$POETRY` below... I think exporting `PATH` might lead to other bugs, but I think it should be fine for now and we can always change it later.